doc/http_methods.rst

```diff
@@ -0,0 +1,221 @@
+
+.. code-block:: python
+
+    examplebuilder = DirectBuilder(metadata_file_contents, restrictions,"PUT")
+
+On calling *SingleQueryable.generate()*, this would return a fuzzed URI string and a JSON string containing all the fuzzed properties. All the keys appear in the URI and are synchronized with the body. 
+
+::
+     
+     Categories(CategoryID=1714953551,CategoryName='%21%C2%9Dla%C3%92l%24',Description='hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1',Picture=binary'ac9916669fAeb2')?sap-client=500   
+     
+     {"CategoryID": "1714953551", "CategoryName": "%21%C2%9Dla%C3%92l%24", "Description": "hz%60%C3%8F%C3%8F%7B%C3%AAi%2Bk%C3%81%C2%A4%C3%96xc%C5%93%C2%A85k%C3%93%2A%C3%B5%C2%BBrLD%2A%E2%80%A1", "Picture": "YmluYXJ5J2FjOTkxNjY2OWZBZWIyJw=="}
+
+
+
+
+
+==================
+Code Documentation
+==================
+
+Adding method parameter in DirectBuilder
+----------------------------------------
+
+DirectBuilder now has an additional parameter called **method**. This accepts the users choice of HTTP method to get the fuzzed requests. The DirectBuilder *init* checks for validity and calls the *Config.fuzzer* setter to set the value in the Config object.
+
+.. code-block:: python
+
+    def __init__(self, metadata, restrictions,method):
+        if method not in ["GET","DELETE","PUT","POST"]:
+            raise ValueError("The http method value \'{}\' is invalid\nUse either GET, DELETE, PUT or POST".format(method))
+        self._queryable = QueryableEntities()
+        self._metadata_string = metadata
+        self._restrictions = restrictions
+        Config.init()
+        Config.fuzzer.http_method_enabled = method
+
+In config.py, this would be used to build the Config object, which would be looked up during query construction and check which HTTP method is set.
+
+Truncating Query Options
+------------------------
+
+For Odata queries other than **GET**, query options need to be truncated. To implement this, in fuzzer.py *Query.build_string()* would first check if *Config.fuzzer.http_method_enabled == "GET"* before generating the options and appending them. 
+The list of options are illustrated in the table above.
+
+
+Generating a Body
+-----------------
+
+For PUT and POST queries, a new element for the queries need to be generated i.e. the **body** of the request. 
+The first change is returning a tuple of *query, body* instead of just the *query* from *SingleQueryable.generate()*.
+A new function *generate_body()* is added which fetches the proprties from the metadata and calls the generator on each of them, and appends them is a dictionary.
+The dictionary is jsonified before being returned as the body. This process is skipped for **GET** and **DELETE**,and they return an empty jsonified string instead, to be compatible with the tuple returned. The changes made to the generators are described further below.
```

Comment:
"the dictionary is jsoified befor being returned as the body" - does this mean that you expects ONLY JSON as possible format? It is possible that your changes were tested only against odata services that produces and consumes just the JSON option. 

But Odata protocol - and odfuzz is trying to cover that- is having both XML/Atom representation and JSON representation. https://www.odata.org/documentation/odata-version-2-0/atom-format/

the CLI configuration options
```
export ODFUZZ_DATA_FORMAT=json
export ODFUZZ_DATA_FORMAT=xml 
```

Check the usage of: 
https://github.com/SAP/odfuzz/blob/bf08492ce8914a56f27a7f7bd3d0e5352064e722/odfuzz/config.py#L47:


@AlapanGhosh Please confirm my suspicion. If yes, I have no problem merging as it currently is, marking another alpha version, but it is another task to cover in some next PR.

Reply:
@phanak-sap yes, currently the body is generated in only JSON. Wont be problem to parse into XML either, and the config already exists, but a setter needs to be added.  Best to have a separate PR.

Reply:
@AlapanGhosh I agree that the scope of the PR is getting bigger and bigger by almost each comment here :)
I have tracked this as another task for you under the https://github.com/SAP/odfuzz/issues/7